### PR TITLE
adds a happy path scenario for EncryptionKeyController

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/encryption/key_controller_encryption_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+							t.Errorf(diff.ObjectDiff(actualSecret, expectedSecret))
+						}
+						wasSecretValidated = true
+						break
+					}
+				}
+				if !wasSecretValidated {
+					t.Errorf("the secret wasn't created and validated")
+				}
+				// TODO: we could actually try to encrypt some data with the key to prove it is valid
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/key_controller_encryption_test.go]: wdyt ?

Comment (p0lyn0mial) [reply pkg/operator/encryption/key_controller_encryption_test.go]: on the other hand, the key contains some random bytes so checking the length of the key should be enough.

File: pkg/operator/encryption/key_controller_encryption_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		initialSecrets           []runtime.Object
+		encryptionSecretSelector metav1.ListOptions
+		targetNamespace          string
+		targetGRs                map[schema.GroupResource]bool
+		validateFunc             func(actions []clientgotesting.Action, targetNamespace string, targetGRs map[schema.GroupResource]bool)
+	}{
+		// scenario 1: assumes a clean slate, that is, there are no previous resources in the system.
+		// It expects that a secret resource with an appropriate key, name and labels will be created.
+		{
+			name: "scenario 1: checks if a secret with AES256 key for core/secret is created",
```
Comment (sttts) [inline pkg/operator/encryption/key_controller_encryption_test.go]: nit: I use not to number scenarios. There will always be another one to insert between existing ones. But just my taste :)

File: pkg/operator/encryption/key_controller_encryption_test.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+)
+
+func TestEncryptionKeyController(t *testing.T) {
+	scenarios := []struct {
+		name                     string
+		initialSecrets           []runtime.Object
+		encryptionSecretSelector metav1.ListOptions
+		targetNamespace          string
+		targetGRs                map[schema.GroupResource]bool
+		validateFunc             func(actions []clientgotesting.Action, targetNamespace string, targetGRs map[schema.GroupResource]bool)
```
Comment (sttts) [inline pkg/operator/encryption/key_controller_encryption_test.go]: if you pass `t` as first arg, you can use `t.Fatal` below.

Comment (sttts) [reply pkg/operator/encryption/key_controller_encryption_test.go]: Oh, you even have to. The `t` bound in these scenario funcs is not the subtest one now.

**Comment by sttts** [comment]
Some nits. Good start!

File: pkg/operator/encryption/key_controller_encryption_test.go (hunk 3 above)
Comment (p0lyn0mial) [reply pkg/operator/encryption/key_controller_encryption_test.go]: good catch, thx!

**Comment by p0lyn0mial** [comment]
comments applied, PTAL

**Comment by enj** [comment]
LGTM, merging so others can add more tests as needed.

**Pull request merged**